### ISSUE TITLE
Possibly incompatible with Google's Room persistence library for Android?

### ISSUE THREAD

**Issue by es0329** [kamikat/moshi-jsonapi#68]
[Room](https://developer.android.com/topic/libraries/architecture/room.html) attempts to match a model's fields with valid [SQLite data types](https://www.sqlite.org/datatype3.html). It will also disregard fields with an [`@Ignore`](https://developer.android.com/reference/android/arch/persistence/room/Ignore.html) annotation. Otherwise, a [type converter](https://developer.android.com/topic/libraries/architecture/room.html#type-converters) must be supplied where type affinity cannot be made by the compiler.

My models subclass [`Resource`](https://github.com/kamikat/moshi-jsonapi/blob/3.2.2/src/main/java/moe/banana/jsonapi2/Resource.java#L11) and [`ResourceIdentifier`](https://github.com/kamikat/moshi-jsonapi/blob/3.2.2/src/main/java/moe/banana/jsonapi2/ResourceIdentifier.java#L15-L18) in turn. I think without annotations or a type converter, those super class fields might trigger the following at compile time:
```
error: Cannot figure out how to save this field into database. You can consider adding a type converter for it.
```

I'm not certain of the issue so I continue trying to understand it but I wanted to share the issue.

**Comment by kamikat** [comment]
There's nothing to do with the library. To implement type converters for every data model is terrible but the only solution. 

I opened the [issue](https://issuetracker.google.com/issues/64951419) to see if it help.

**Comment by es0329** [comment]
At the time of our inquiry, I moved forward by separating DTO and DAO. Now, [@Ignore superclass fields](https://issuetracker.google.com/issues/63522075) released in [Room v2.1.0-alpha01](https://developer.android.com/jetpack/docs/release-notes#october_8_2018), looks a lot like what we requested.

Separate models was appropriate for my case but I wanted to update our issue for future reference.
